packages/vscode-extension/src/panels/webviewContentGenerator.ts

```diff
@@ -2,40 +2,48 @@ import { ExtensionContext, ExtensionMode, Webview, Uri, workspace } from "vscode
 import { getUri } from "../utilities/getUri";
 import { getNonce } from "../utilities/getNonce";
 import { Platform } from "../utilities/platform";
-
-const VITE_DEV_HOST = "localhost:2137";
-
-const VITE_DEV_PREABMLE = /*html*/ `
-<script type="module">
-import RefreshRuntime from "/@react-refresh"
-RefreshRuntime.injectIntoGlobalHook(window)
-window.$RefreshReg$ = () => {}
-window.$RefreshSig$ = () => (type) => type
-window.__vite_plugin_react_preamble_installed__ = true
-</script>
-<script type="module" src="/@vite/client"></script>
-`;
+import { EXTENSION_DEV_SERVER_HOST } from "../webview/utilities/constants";
 
 export function generateWebviewContent(
   context: ExtensionContext,
   webview: Webview,
-  extensionUri: Uri
+  extensionUri: Uri,
+  webviewName: string,
```

Comment:
I think it'd be cleaner if we place `network` code under `src/network` and not `src/webview` which was meant to be a folder that's dedicated to the main webview panel (I know the name isn't ideal).

I'd recommend that we move the network code to `src/network` and that we pass a single parameter here `moduleName` from which we will derive the dev path `src/${moduleName}/index.jsx`, the release bundle path `dist/${moduleName}.js` and the CSS entrypoint `dist/${moduleName}.css`